packages/headless/src/controllers/generated-answer/headless-generated-answer.ts

```diff
@@ -25,14 +29,20 @@ export type {
  * @returns A `GeneratedAnswer` controller instance.
  */
 export function buildGeneratedAnswer(
-  engine: SearchEngine,
+  engine: SearchEngine | EngineNeededForKnowledgeGeneratedAnswer,
```

Comment:
If you apply the comments I left here https://github.com/coveo/ui-kit/pull/4129/files#diff-644461440e8a66297a77b95e54021c1e4840bad4c475302dea37bfc17945091cR117, then this would be

```suggestion
  engine: SearchEngine | InsightEngine | EngineNeededForKnowledgeGeneratedAnswer,
```